src/components/MatchRuleList.tsx

```diff
@@ -0,0 +1,255 @@
+    setMatchRules([
+      { id: matchRules.length.toString(), links: {} },
+      ...matchRules,
+    ])
+  }
+
+  const deleteMatchRule = (id: string) => {
+    console.log(id)
+    let newMatchRules = [...matchRules]
+    const ruleIndex = matchRules.findIndex(rule => rule.id === id)
+
+    newMatchRules[ruleIndex].toDelete = true
+    console.log(newMatchRules)
+    setMatchRules(newMatchRules)
+  }
+
+  const saveMatchRules = async () => {
+    let updatedMatchRules: MatchRule[] = []
+
+    const requests = matchRules.map((rule, index) => {
+      // If the rule should be deleted, we delete it. This is done before checking validity, because
+      // a rule might have been edited to be invalid and then deleted
+      //
+      // Rules that have not been created yet are also skipped, since they will be gone with the re-render
+      if (rule.toDelete && rule.createdAt) {
+        return matchRuleApi.delete(rule).catch(err => setError(err.message))
+      }
+
+      // If the rule is missing an account ID or a match, it is invalid, so we discard it
+      if (!rule.accountId || !rule.match) {
+        return true
+      }
+
+      if (rule.createdAt) {
+        // createdAt is set by the backend, so this is an existing rule that needs to be updated
+        rule.priority = index
+
+        return matchRuleApi
+          .update(rule)
+          .then(updatedRule => {
+            updatedMatchRules.push(updatedRule)
+          })
+          .catch(err => setError(err.message))
+      } else {
+        // This is a new rule, it needs to be created
+        rule.priority = index
+        return matchRuleApi
+          .create(rule, budget)
+          .then(newRule => {
+            updatedMatchRules.push(newRule)
+          })
+          .catch(err => setError(err.message))
+      }
+    })
+
+    await Promise.all(requests)
+
+    setMatchRules(updatedMatchRules.sort((a, b) => a.priority - b.priority))
+    setNotification(t('changesSaved'))
+  }
+
+  useEffect(() => {
+    if (!isLoading) {
+      setIsLoading(true)
+    }
+
+    const promises = [
+      matchRuleApi.getAll(budget).then(setMatchRules),
+      accountApi.getAll(budget).then(setAccounts),
+    ]
+
+    Promise.all(promises)
+      .then(() => setError(''))
+      .catch(err => {
+        setError(err.message)
+      })
+      .then(() => setIsLoading(false))
+  }, [budget])
+
+  return (
+    <>
+      <div className="header mb-4">
+        <h1>{t('matchRules.matchRules')}</h1>
+
+        <div className="header--action full-centered">
+          <button onClick={() => setHideHelp(!hideHelp)} title={t('help')}>
+            <QuestionMarkCircleIcon className="icon-sm" />
+          </button>
+
+          <button onClick={() => addMatchRule()} title={t('matchRules.add')}>
+            <PlusIcon className="icon-sm" />
+          </button>
+
+          <button onClick={() => saveMatchRules()} title={t('save')}>
+            <CloudArrowUpIcon className="icon-sm" />
```

Comment:
This is not very intuitive as a save button. I would suggest something like this:
![image](https://github.com/user-attachments/assets/7240fcfd-d8c0-419f-8872-b74011e5b231)
mobile layout:
![image](https://github.com/user-attachments/assets/e80114ad-207a-4c85-ab07-c9c89b97b024)

Reply:
Implemented in 80745c0b64c69f5452a0f0d267f13f1b1b76045e